src/RageSoundPosMap.cpp

```diff
@@ -54,71 +57,53 @@ pos_map_queue &pos_map_queue::operator=( const pos_map_queue &rhs )
 	return *this;
 }
 
-void pos_map_queue::Insert( std::int64_t iSourceFrame, int iFrames, std::int64_t iDestFrame, float fSourceToDestRatio )
+void pos_map_queue::Insert(std::int64_t iSourceFrame, std::int64_t iFrames, std::int64_t iDestFrame, double fSourceToDestRatio)
 {
-	if( !m_pImpl->m_Queue.empty() )
+	bool merged = false;
+	if (!m_pImpl->m_Queue.empty())
 	{
-		/* Optimization: If the last entry lines up with this new entry, just merge them. */
-		pos_map_t &last = m_pImpl->m_Queue.back();
-		if( last.m_iSourceFrame + last.m_iFrames == iSourceFrame &&
-		    last.m_fSourceToDestRatio == fSourceToDestRatio &&
-		    llabs(last.m_iDestFrame + std::lrint(last.m_iFrames * last.m_fSourceToDestRatio) - iDestFrame) <= 1 )
-		{
-			last.m_iFrames += iFrames;
-
-			/* Make sure that m_Frames doesn't grow too large and overflow an int. */
-			if( !m_pImpl->m_Queue.empty() && last.m_iFrames > pos_map_backlog_frames * 2 )
```

Comment:
Do we no longer have to do any of the frame splitting? Is it because we assume m_iFrames isn't going to ever get that large?

Reply:
This is correct. Originally there was a mix of 32 bit and 64 bit integers in this code, so this was needed to prevent an integer overflow -  if the value started getting large, it would be split up to two int's and those would be processed separately. Now that all variables are 64-bit here, there is no need for this code. (Presumably this was a nice performance trick for 2003, when the code needed to run on a Pentium II and avoiding 64 bit integers where possible was important.)